src/simplify_reshapes.cpp

```diff
@@ -678,9 +678,12 @@ struct find_transpose_contiguous_reshaper_unary
         auto trans_ins     = r.instructions["trans_ins"];
         auto cont_ins      = r.instructions["cont_ins"];
         auto unary_op_name = ins->get_operator().name();
-        auto unary_ins     = m.insert_instruction(cont_ins, make_op(unary_op_name), trans_ins);
```

Comment:
This should be like 
`make_op(unary_op_name, ins->get_operator().to_value())`

Reply:
@umangyadav  Yes, I will fix it to this more general solution and update the PR description.

Reply:
can you add a unit- test ?

Reply:
@umangyadav I have added a unit test that represents the scenario that caused the issue in the model. It fails without the fix.